### DISCUSSION TITLE
mirrord 3.1.0 Feedback Thread

### DISCUSSION THREAD

**Discussion by eyalb181** [metalbear-co/mirrord#598] [General]
Hey everyone,

We've just launched version 3.1.0, which is our stable version to provide the full mirrord experience - incoming/outgoing network traffic, file access, and environment variables all plugged into the cloud, wrapping your local process in the full context of your Kubernetes cluster.

It's important for us to hear about your experience with this new version - what worked, what didn't, what problems it solves for you and what features you'd like to see next. Please feel free to drop a reply in the thread (you can also open an issue or chat with us on [Discord](https://discord.gg/pSKEdmNZcK) if you prefer).

Happy mirroring!
Eyal

**Comment by martaver** [comment]
Hi,

We're considering using mirrord in our stack, so I've been following it for a few releases now. 

I see it's now possible to impersonate at the Deployment / Pod / Container levels. This is an important development for us.

My feedback is that it's difficult to tell from your website/documentation copy:
- that the above is possible
- whether impersonating multiple services at once is possible
- whether it's possible to only impersonate traffic that is identified by header

Our use case is a monorepo - where we would impersonate each changed service for testing purposes. We would also need to do so in such a way that the testing traffic for various developers is isolated (hence the header).

**Reply by aviramha** [reply]
Hey! Thanks for the feedback, it's extremely valuable for us.

- You can impersonate a deployment, at the moment it will choose one pod (out of the running replicas/pods) to mirror/steal traffic from.
- Re Impersonating multiple services
  - if you mean running many services locally, each impersonating a different deployment/pod remotely, then yes
  - If you mean running one service locally, impersonating a deployment so it receives traffic from all of its pods, currently no. We're trying to see if there's enough interest, you can upvote it here (and also follow for updates): #541.
- Conditional filtering (aka header) - it's not currently supported, but in the roadmap. You can follow #290 for updates.

We're currently researching the use cases for conditional filtering/deployment impersonation - trying to see how different people would use them so we can design them more accurately. If you happen to have the time, we'd love to have a quick chat to learn more about your specific use case.
You can schedule [here](https://calendly.com/d/drb-fg3-8m2/metalbear-meeting).
If you prefer async chat, we can take it on Discord, or also here. The meeting is just so we can do it at a higher bandwidth :)
